src/server-glue/module.ts

```diff
@@ -72,6 +75,10 @@ export const init = async <T extends string>(
 		}
 		await Promise.all(promises);
 
+		if (process.env.PINEJS_STORAGE_ENGINE === 'S3') {
+			initS3Storage();
```

Comment:
initialization can fail because of invalid parameters or failure to create an S3Client. In any case, since the configuration won't succeed, the error will be matched by the handler below and the initialization will abort:

Reply:
Moved the initialization up in the code.